compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.cpp

```diff
@@ -92,6 +92,76 @@ SizesAndScalableFlags TilingConfig::getVectorTileSizes() {
   return std::make_pair(vectorSizes, scalableFlags);
 }
 
+IREE::Codegen::LoweringConfigAttr
+TilingConfig::withNewVectorSizes(ArrayRef<int64_t> newSizes,
+                                 ArrayRef<bool> newScalableFlags) {
+  unsigned numDims = getNumDimensions();
+  assert(newSizes.size() == numDims &&
+         "expected `sizes` to match number of dimensions");
+  assert((newScalableFlags.empty() || newScalableFlags.size() == numDims) &&
+         "expected `scalableFlags` to match "
+         "number of dimensions (or be empty)");
+
+  auto tilingLevels = loweringConfig.getTilingLevels();
+  auto tilingLevelHasSize = [tilingLevels](unsigned tilingLevelIndex,
+                                           unsigned dimPos) mutable {
+    auto level = tilingLevels[tilingLevelIndex];
+    return level.getSizes()[dimPos] != 0;
+  };
```

Comment:
```suggestion
  bool isDimSizeNonZeroForTilingLevel = [tilingLevels](unsigned tilingLevelIndex,
                                           unsigned dimPos) mutable {
    auto level = tilingLevels[tilingLevelIndex];
    return level.getSizes()[dimPos] != 0;
  };
```

It's a lambda rather than a function, but still feels that following this makes sense: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly